tests/unit/test_images.py

```diff
@@ -0,0 +1,148 @@
+import os
+import uuid
+from unittest import mock
+
+import jsonschema
+import pytest
+
+from app.services.images import ImageService
+
+TEST_IMAGE = "test_cat.jpg"
+TEST_IMAGE_PATH = f"{os.path.dirname(os.path.realpath(__file__))}/assets/{TEST_IMAGE}"
+TEST_UUID = uuid.UUID("c0ff33c0-ffee-c0ff-eec0-ffeec0ffeec0")
+QUERY = "flowers"
+
+DOWNLOADED_PATH = ImageService.get_downloaded_path()
+DOWNLOADED_FILE_PATH = f"{DOWNLOADED_PATH}/{QUERY}/{str(TEST_UUID)}.jpg"
+UPLOAD_PATH = ImageService.get_upload_path()
+UPLOAD_FILE_PATH = f"{UPLOAD_PATH}/{TEST_IMAGE}"
+
+
+# Validation Schemas
+model_prediction_schema = {
+    "type": "object",
+    "properties": {
+        "class_name": {"type": "string"},
+        "confidence": {"type": "number", "minimum": 0, "maximum": 1.0},
+    },
+    "required": ["class_name", "confidence"],
+    "additionalProperties": False,
+}
+
+model_layer_schema = {
+    "type": "object",
+    "properties": {
+        "name": {"type": "string"},
+        "class_name": {"type": "string"},
+        "params": {"type": "integer"},
+        "output_shape": {"type": "array"},
+    },
+    "required": ["name", "class_name", "params", "output_shape"],
+    "additionalProperties": False,
+}
+
+model_summary_schema = {
+    "type": "object",
+    "properties": {
+        "name": {"type": "string"},
+        "total_params": {"type": "integer"},
+        "trainable_params": {"type": "integer"},
+        "non_trainable_params": {"type": "integer"},
+        "layers": {
+            "type": "array",
+            "items": model_layer_schema,
+        },
+    },
+    "required": [
+        "name",
+        "total_params",
+        "trainable_params",
+        "non_trainable_params",
+        "layers",
+    ],
+    "additionalProperties": False,
+}
+
+
+# Fixtures
+@pytest.fixture()
+def twitter_image_crawler():
+    crawler_mock = mock.Mock()
+    crawler_mock.fetch_urls_by_query.return_value = [
+        # TODO: may not be very stable--host images or mock.
+        "https://fastly.picsum.photos/id/908/300/300.jpg?hmac=Kr9iK9ySwwpN-5dnU7po_rUxEoeoaJrp0bjX21M1sd4"
+    ]
+    with mock.patch(
+        "app.routers.images.TwitterImageCrawler", return_value=crawler_mock
+    ):
+        yield crawler_mock
+
+
+@pytest.fixture()
+def static_uuid():
+    # Remove image from previous runs.
+    try:
+        os.remove(DOWNLOADED_FILE_PATH)
+    except OSError:
+        pass
+
+    with mock.patch("app.services.images.uuid4", return_value=TEST_UUID):
+        yield TEST_UUID
```

Comment:
tests will actually check images were actually uploaded/downloaded to file system. Deletes these test artifacts before a run (not after, in case one may want to inspect the assets resulting from the test run, directly)